app/questionnaire/rules.py

```diff
@@ -116,11 +116,18 @@ def evaluate_repeat(repeat_rule, answer_store, answer_ids_on_path):
         answers = list(answer_store.filter(answer_ids=[when['id']]))
         stop = bool(answers and evaluate_rule(when, answers[-1]['value']))
         no_of_repeats = len(answers) + (0 if stop else 1)
+    elif repeat_rule['type'] == 'group':
+        group_instances = set()
+        for group_id in repeat_rule['group_ids']:
+            answer_ids_in_group = schema.get_answer_ids_for_group(group_id)
+            group_instances |= {a['group_instance_id'] for a in answer_store.filter(answer_ids=answer_ids_in_group)}
+
+        no_of_repeats = len(group_instances)
     else:
         repeat_functions = {
             'answer_value': _get_answer_value,
-            'answer_count': lambda f: _get_answer_count(f, repeat_rule),
-            'answer_count_minus_one': lambda f: min(_get_answer_count_minus_one(f), MAX_REPEATS - 1),
+            'answer_count': len,
```

Comment:
How is this working?  There's no variable called `len` in this module

Reply:
there is a function in python called `len` thou

Reply:
Yea, but when you call it you have to give it an object don't you? unless there's some hidden 'when called without arguments, it counts the length of the last functions return' or something?